worker/src/ipfs.rs

```diff
@@ -68,14 +65,15 @@ pub fn read_from_ipfs(cid: Cid) -> Vec<u8> {
 
     let (tx, rx) = channel();
 
-    let req = client
-        .cat(h)
-        .concat2()
-        .map(move |res| {
-            tx.send(res).unwrap();
-        })
-        .map_err(|e| eprintln!("{}", e));
-    hyper::rt::run(req);
+    match client.cat(h)
+        .map_ok(|chunk| chunk.to_vec())
+        .try_concat()
+		.await
+		{
+            Ok(res) => { tx.send(res).unwrap(); },
+            Err(e) => eprintln!("error reading file: {}", e)
+        }
+
     rx.recv().unwrap().to_vec()
```

Comment:
Just a short question, do you have any idea, whether these `tx.send` and `rx.recv` unwraps can panic?

If so, I would like you to put some graceful error handling there. An enclave must never panic as this would be a potential security issue. 

I know they have been there before, but this was unused code.

Reply:
feel free to change the return types of these functions to `Results<>` if you see fit.

Reply:
Nevermind, I will merge this. Then we have a simple update PR. You will probably touch this code anyhow in your remaining work.